middle_end/flambda2/flambda_middle_end.ml

```diff
@@ -82,47 +82,50 @@ let middle_end0 ppf ~prefixname ~backend ~filename ~module_ident
     ~module_block_size_in_words ~module_initializer =
   Misc.Color.setup (Flambda_features.colour ());
   Profile.record_call "flambda.0" (fun () ->
-      let flambda =
+      let flambda, code =
         Profile.record_call "lambda_to_flambda" (fun () ->
             Lambda_to_flambda.lambda_to_flambda ~backend ~module_ident
               ~module_block_size_in_words module_initializer)
       in
       print_rawflambda ppf flambda;
       check_invariants flambda;
-      let flambda =
-        if Flambda_features.Debug.permute_every_name ()
-        then Flambda_unit.permute_everything flambda
-        else flambda
-      in
-      let round = 0 in
-      let new_flambda =
-        Profile.record_call ~accumulate:true "simplify" (fun () ->
-            Simplify.run ~backend ~round flambda)
-      in
-      (if Flambda_features.inlining_report ()
-      then
-        let output_prefix = Printf.sprintf "%s.%d" prefixname round in
-        Inlining_report.output_then_forget_decisions ~output_prefix);
-      print_flambda "simplify" ppf new_flambda.unit;
-      output_flexpect ~ml_filename:filename flambda new_flambda.unit;
-      new_flambda)
+      if Flambda_features.Expert.fallback_inlining_heuristic ()
+      then { cmx = None; unit = flambda; all_code = code }
```

Comment:
Just to make sure, but will this create spurious warnings related to `-opaque` ? I.e. the compiler can complain when it doesn't find the `.cmx` of a compilation unit not compiled with `-opaque` iirc, so the question is, since there is `None` here for `cmx`, will this emit an empty `.cmx` file, or will this not emit a `.cmx` file at all, in which case, will this trigger any additional warning ?

Small aside: maybe we should add in the testsuite some tests that are run at various optimizations levels to verify that  some optimizations levels do not break some things ?

Reply:
I did run the testsuite with `OCAMLPARAM=_,Oclassic=1`, and I think I also tried on the compilation itself at some point (on the previous repository) and I don't recall seeing the issue you described.
I believe all the failures in the testsuite in classic mode that I saw can be explained by the lack of transformations (e.g. symbol lifting).

Reply:
There are currently no warnings in Flambda 2 for missing cmx files. We will probably implement them at some point, but for now `None` is the correct value to pass in this case.
A `.cmx` file will always be generated (it contains more than just the approximations), but when either `-opaque` is passed or we can't generate an approximation (this can happen when the module initialisation code never returns), then the approximations field of the cmx is set to a dummy value.